howso/hyperparameters.amlg

```diff
@@ -225,48 +225,105 @@
 				(set !hyperparameterMetadataMap (list action_feature context_features_key robust_mode weight_feature) baseline_hyperparameter_map)
 		))
 
-		;if deviations were computed, store them in !residualsMap
-		(if deviations
-			(assign_to_entities (assoc
-				!residualsMap
-					(set
-						!residualsMap
-						(concat robust_mode action_feature robust_mode weight_feature)
-						(append
-							deviations
-							(assoc
-								".robust" (= "robust" robust_mode)
-								".hyperparam_path" (list action_feature context_features_key robust_mode weight_feature)
+		(if bypass_calculate_prediction_stats
+			;if deviations were computed, store them in !residualsMap
+			(if deviations
+				(assign_to_entities (assoc
+					!residualsMap
+						(set
+							!residualsMap
+							(concat robust_mode action_feature robust_mode weight_feature)
+							(append
+								deviations
+								(assoc
+									".robust" (= "robust" robust_mode)
+									".hyperparam_path" (list action_feature context_features_key robust_mode weight_feature)
+								)
 							)
 						)
+				))
+
+				;else compute and cache residuals and featureDomainAttributes if there are features with nulls
+				(let
+					(assoc
+						num_features_with_nulls
+							(size (filter
+								;keep only features that have nulls
+								(lambda (get (current_value) "has_nulls") )
+								;ignore all inactive features (where all values are nulls)
+								(remove !featureNullRatiosMap (indices !inactiveFeaturesMap))
+							))
 					)
-			))
 
-			;else compute and cache residuals and featureDomainAttributes if there are features with nulls
-			(let
-				(assoc
-					num_features_with_nulls
-						(size (filter
-							;keep only features that have nulls
-							(lambda (get (current_value) "has_nulls") )
-							;ignore all inactive features (where all values are nulls)
-							(remove !featureNullRatiosMap (indices !inactiveFeaturesMap))
+					(if (> num_features_with_nulls 0)
+						(call !CalculateAndStoreFeatureResiduals (assoc
+							features (values (append context_features action_features) (true))
+							robust_residuals (= "robust" robust_mode)
+							hyperparameter_feature action_feature
+							weight_feature weight_feature
+							use_case_weights use_case_weights
+							custom_hyperparam_map baseline_hyperparameter_map
+							compute_all_statistics (false)
 						))
-				)
-
-				(if (> num_features_with_nulls 0)
-					(call !CalculateAndStoreFeatureResiduals (assoc
-						features (values (append context_features action_features) (true))
-						robust_residuals (= "robust" robust_mode)
-						hyperparameter_feature action_feature
-						weight_feature weight_feature
-						use_case_weights use_case_weights
-						custom_hyperparam_map baseline_hyperparameter_map
-					))
 
+					)
 				)
 			)
+			(call !CalculateAndStoreFeatureResiduals (assoc
```

Comment:
add a comment here saying ;else compute and cache residuals and stats using most current hyperparameters